app/managers/banner_manager.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {Screens} from '@constants';
+import {showOverlay, dismissOverlay} from '@screens/navigation';
+import {toMilliseconds} from '@utils/datetime';
+
+import type {BannerConfig} from '@components/floating_banner/types';
+
+const FLOATING_BANNER_OVERLAY_ID = 'floating-banner-overlay';
+const TIME_TO_CLOSE = toMilliseconds({seconds: 5});
+
+class BannerManagerSingleton {
+    private isVisible = false;
+    private closeTimeout: NodeJS.Timeout | null = null;
+    private currentBannerId: string | null = null;
+    private currentOnDismiss: (() => void) | null = null;
+
+    private clearTimeout(timeout: NodeJS.Timeout | null) {
+        if (timeout) {
+            clearTimeout(timeout);
+        }
+    }
+
+    private invokeCurrentOnDismiss() {
+        if (typeof this.currentOnDismiss === 'function') {
+            try {
+                this.currentOnDismiss();
+            } catch {
+                // Silent catch to ensure cleanup still runs
+            }
+        }
+        this.currentOnDismiss = null;
+    }
+
+    showBanner(bannerConfig: BannerConfig) {
+        if (this.isVisible) {
+            this.clearTimeout(this.closeTimeout);
+            dismissOverlay(FLOATING_BANNER_OVERLAY_ID);
+            this.isVisible = false;
+            this.currentBannerId = null;
+        }
+
+        this.isVisible = true;
+        this.currentBannerId = bannerConfig.id;
+
+        this.currentOnDismiss = bannerConfig.onDismiss || null;
+
+        const handleDismiss = () => {
+            this.isVisible = false;
+            this.currentBannerId = null;
+
+            this.invokeCurrentOnDismiss();
+            dismissOverlay(FLOATING_BANNER_OVERLAY_ID);
+        };
+        let customComponent = bannerConfig.customComponent;
+        if (customComponent && React.isValidElement(customComponent)) {
+            const props: Partial<Record<string, unknown>> = {
+                onDismiss: handleDismiss,
+                dismissible: bannerConfig.dismissible,
+            };
+            customComponent = React.cloneElement(customComponent, props);
```

Comment:
I'm not sure I understand -- why do we need to clone the customComponent? Isn't it content only? The Banner wrapper handles the onDismiss and dismissible, right?